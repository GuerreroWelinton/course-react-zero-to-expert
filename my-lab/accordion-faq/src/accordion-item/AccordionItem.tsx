import { useState } from 'react';
import type { Question } from '../Accordion.interface';
import styles from './AccordionItem.module.css';

const AccordionItem = ({ id, content, title }: Question) => {
  const [isOpen, setIsOpen] = useState(false);

  const onClick = () => {
    setIsOpen(!isOpen);
  };

  const className = `${styles.content} ${isOpen ? styles.open : styles.close}`;
  return (
    <>
      <div className={styles.accordion} onClick={onClick}>
        <h1 className={styles.question}>
          {id}. {title}
        </h1>
        <div className={className}>
          <p className={styles.answer}>{content}</p>
        </div>
      </div>
    </>
  );
};

export default AccordionItem;
