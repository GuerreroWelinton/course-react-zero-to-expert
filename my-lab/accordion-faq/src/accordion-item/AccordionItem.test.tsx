import { render, screen } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import AccordionItem from './AccordionItem';
import type { FAQItem } from '../Accordion.interface';

const faqItem: FAQItem = {
  id: 123,
  question: 'Esta es la pregunta número 1',
  answer: 'Esta es la respuesta a la pregunta número 1',
  customBgColor: '#acc754',
};

describe('AccordionItem', () => {
  test('should render with default values', () => {
    render(<AccordionItem {...faqItem} />);

    expect(screen.getByText(faqItem.answer)).toBeDefined();
  });

  test('should match the snapshot', () => {
    const { container } = render(<AccordionItem {...faqItem} />);

    expect(container).toMatchSnapshot();
  });
});
