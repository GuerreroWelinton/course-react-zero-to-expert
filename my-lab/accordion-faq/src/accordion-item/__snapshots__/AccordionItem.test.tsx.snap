// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AccordionItem > should match the snapshot 1`] = `
<div>
  <div
    class="_accordion_38baf4"
    data-testid="accordionItem"
    style="background-color: rgb(172, 199, 84);"
  >
    <h1
      class="_question_38baf4"
    >
      123
      . 
      Esta es la pregunta número 1
    </h1>
    <div
      class="_content_38baf4 _close_38baf4"
      data-testid="answer"
    >
      <p
        class="_answer_38baf4"
      >
        Esta es la respuesta a la pregunta número 1
      </p>
    </div>
  </div>
</div>
`;
